### ISSUE TITLE
About the integration

### ISSUE THREAD

**Issue by kissstar11** [suriyun-production/mmorpg-kit-docs#1235]
Hi,I just found the integration of the MMORPG kit and the Fantasy Heroes: 4-Directional Character Editor:
https://github.com/insthync/UnityMultiplayerARPG_HeroEditor4D

1.So can it still work on the latest MMORPG version (1.71)?
2.what steps should I follow to do the integration?

Thank you!

**Comment by insthync** [comment]
1. Yes, try the demo
2. Have to import [this](https://assetstore.unity.com/packages/2d/characters/fantasy-heroes-4-directional-character-editor-epic-extension-147364) and [this](https://github.com/insthync/UnityMultiplayerARPG_HeroEditor4D). Then new settings will be appeared, see what I did to the new settings in the Demo.

**Comment by kissstar11** [comment]
> 1. Yes, try the demo
> 2. Have to import [this](https://assetstore.unity.com/packages/2d/characters/fantasy-heroes-4-directional-character-editor-epic-extension-147364) and [this](https://github.com/insthync/UnityMultiplayerARPG_HeroEditor4D). Then new settings will be appeared, see what I did to the new settings in the Demo.

Thank you, will try it

**Comment by kissstar11** [comment]
> 1. Yes, try the demo
> 
> 2. Have to import [this](https://assetstore.unity.com/packages/2d/characters/fantasy-heroes-4-directional-character-editor-epic-extension-147364) and [this](https://github.com/insthync/UnityMultiplayerARPG_HeroEditor4D). Then new settings will be appeared, see what I did to the new settings in the Demo.

Hi,I imported both two assets and found 2 issues.how to fix them?![image](https://user-images.githubusercontent.com/72058863/140596229-3bf2f804-0d5e-4203-a0df-4085a447e374.jpeg)

![image](https://user-images.githubusercontent.com/72058863/140596236-31e67863-a4c8-4380-a336-0c9d235a2f9b.jpeg)

**Comment by insthync** [comment]
Did you import hero 4d package?

**Comment by kissstar11** [comment]
> Did you import hero 4d package?

Yes I do. I imported the 4d pack before importing the integration files. Is there a path requirement of the integration files？

**Comment by VEsttas** [comment]
> 1. Yes, try the demo
> 2. Have to import [this](https://assetstore.unity.com/packages/2d/characters/fantasy-heroes-4-directional-character-editor-epic-extension-147364) and [this](https://github.com/insthync/UnityMultiplayerARPG_HeroEditor4D). Then new settings will be appeared, see what I did to the new settings in the Demo.

I got this issue too by following the steps.

**Comment by insthync** [comment]
I will recheck again, I might use the old version of Hero Editor 4D

**Comment by kissstar11** [comment]
> I will recheck again, I might use the old version of Hero Editor 4D

Thank you for the support

**Comment by insthync** [comment]
Try delete `Assets/HeroEditor4D/Common/Newtonsoft folder

**Comment by kissstar11** [comment]
> Try delete `Assets/HeroEditor4D/Common/Newtonsoft` folder

I tried that and comes out more issues 🙃
![image](https://user-images.githubusercontent.com/72058863/140601658-511bda4d-0f08-4354-aff4-c578fe4c9b93.jpeg)

**Comment by insthync** [comment]
Okay, bring it back, I don't get an error like yours, what is your Unity version?

**Comment by kissstar11** [comment]
> Okay, bring it back, I don't get an error like yours, what is your Unity version?

2020.3.20    But I am now downloading the 2021 version to try it again.

**Comment by insthync** [comment]
No you don't have to do it

**Comment by insthync** [comment]
Seems like it has a different package version of HeroEditor4D for each Unity version, I use Unity 2019 for development, when I try 2020 I got problems like you too, it is not hard to fix, just have to change namespace.

**Comment by kissstar11** [comment]
> Seems like it has a different package version of HeroEditor4D for each Unity version, I use Unity 2019 for development, when I try 2020 I got problems like you too, it is not hard to fix, just have to change namespace.

Yes, 4D is now only support 2020 and higher version

**Comment by kissstar11** [comment]
> Seems like it has a different package version of HeroEditor4D for each Unity version, I use Unity 2019 for development, when I try 2020 I got problems like you too, it is not hard to fix, just have to change namespace.

I would appreciate it if you could update the integration since I am really new in UNITY. Thank you！

**Comment by insthync** [comment]
Updated

**Comment by kissstar11** [comment]
> Updated

Hi there,sorry for disturbing you but I still got issues when I tried the new update files. New errors occurred when I created the character in the 00init_hero 4D scene(the image of the character flashed past down and disappeared and then the errors occurred)
![image](https://user-images.githubusercontent.com/72058863/140618393-b2437f16-aa0e-43d8-a4c2-b000ce53681f.png)

![image](https://user-images.githubusercontent.com/72058863/140618403-68394935-7f76-4d23-af3b-d8148d57b5f8.png)

UnassignedReferenceException: The variable CapeRenderer of Character has not been assigned.
You probably need to assign the CapeRenderer variable of the Character script in the inspector.
UnityEngine.SpriteRenderer.get_color () (at <67bbede196d54d91bb0ca9922511a90c>:0)
Assets.HeroEditor4D.Common.CharacterScripts.Character.Equip (HeroEditor4D.Common.SpriteGroupEntry item, HeroEditor4D.Common.Enums.EquipmentPart part, System.Nullable`1[T] color) (at Assets/HeroEditor4D/Common/CharacterScripts/CharacterSetup.cs:172)
Assets.HeroEditor4D.Common.CharacterScripts.Character.Equip (HeroEditor4D.Common.SpriteGroupEntry item, HeroEditor4D.Common.Enums.EquipmentPart part) (at Assets/HeroEditor4D/Common/CharacterScripts/CharacterSetup.cs:195)
Assets.HeroEditor4D.Common.CharacterScripts.Character.UnEquip (HeroEditor4D.Common.Enums.EquipmentPart part) (at Assets/HeroEditor4D/Common/CharacterScripts/CharacterSetup.cs:203)
HeroEditor4D.Common.Character4DBase+l.A (HeroEditor4D.Common.CharacterBase ) (at <92a966cee9624c60b476ce362feab14a>:0)
System.Collections.Generic.List`1[T].ForEach (System.Action`1[T] action) (at <695d1cc93cca45069c528c15c9fdd749>:0)
HeroEditor4D.Common.Character4DBase.UnEquip (HeroEditor4D.Common.Enums.EquipmentPart part) (at <92a966cee9624c60b476ce362feab14a>:0)
MultiplayerARPG.HeroEditor4D.HeroEditorModel.SetEquipmentSprites () (at Assets/UnityMultiplayerARPG_HeroEditor4D-main/Scripts/CharacterModel/HeroEditorModel.cs:178)
MultiplayerARPG.HeroEditor4D.HeroEditorModel.SetEquipWeapons (MultiplayerARPG.EquipWeapons equipWeapons) (at Assets/UnityMultiplayerARPG_HeroEditor4D-main/Scripts/CharacterModel/HeroEditorModel.cs:86)
MultiplayerARPG.BaseCharacterModel.SwitchModel (MultiplayerARPG.BaseCharacterModel previousModel) (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Model/BaseCharacterModel.cs:252)
MultiplayerARPG.CharacterModelManager.SwitchTpsModel (MultiplayerARPG.BaseCharacterModel nextModel) (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Model/CharacterModelManager.cs:151)
MultiplayerARPG.CharacterModelManager.EntityAwake () (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Model/CharacterModelManager.cs:66)
MultiplayerARPG.BaseGameEntity.Awake () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BaseGameEntity.cs:276)
UnityEngine.Object:Instantiate(BaseCharacterEntity, Transform)
MultiplayerARPG.CharacterDataExtension:InstantiateModel(ICharacterData, Transform) (at Assets/UnityMultiplayerARPG/Core/Scripts/CharacterData/CharacterDataExtension.cs:55)
MultiplayerARPG.UICharacterList:<LoadCharacters>b__23_0(Int32, PlayerCharacterData, GameObject) (at Assets/UnityMultiplayerARPG/Core/Scripts/LanGame/UI/UICharacterList.cs:113)
UIList:Generate(IEnumerable`1, Action`3) (at Assets/UnityMultiplayerARPG/Core/Scripts/UI/UIList.cs:54)
MultiplayerARPG.UICharacterList:LoadCharacters() (at Assets/UnityMultiplayerARPG/Core/Scripts/LanGame/UI/UICharacterList.cs:104)
MultiplayerARPG.UICharacterList:OnEnable() (at Assets/UnityMultiplayerARPG/Core/Scripts/LanGame/UI/UICharacterList.cs:154)
UnityEngine.GameObject:SetActive(Boolean)
UIBase:Show() (at Assets/UnityMultiplayerARPG/Core/Scripts/UI/UIBase.cs:64)
UIHistory:Next(UIBase) (at Assets/UnityMultiplayerARPG/Core/Scripts/UI/UIHistory.cs:27)
MultiplayerARPG.UISceneHome:OnClickSinglePlayer() (at Assets/UnityMultiplayerARPG/Core/Scripts/LanGame/UI/Scenes/UISceneHome.cs:17)
UnityEngine.EventSystems.EventSystem:Update() (at Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/EventSystem.cs:385)
